### PR TITLE
UPSTREAM: <carry>: openshift: Openshift fixups, dockerfile and labels

### PR DESCRIPTION
**Pull request by awesomenix**
**What this PR does / why we need it**:
- Build both openshift cluster api manager and manager hosting azure provider, fixing docker file
- Change all machine role labels to refer openshift ones
- Revendor all changes

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -926,7 +926,7 @@
   version = "kubernetes-1.13.1"
```
Comment (enxebre) [inline Gopkg.lock]: all vendor changes does not belong to this commit

Comment (awesomenix) [reply Gopkg.lock]: moved to different commit

File: pkg/cloud/azure/actuators/machine/actuator_test.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ func newFakeScope(t *testing.T, label string) *actuators.MachineScope {
 	}
 	scope.Network().APIServerIP.DNSName = "DummyDNSName"
 	labels := make(map[string]string)
-	labels["set"] = label
+	labels["machine.openshift.io/cluster-api-machine-role"] = label
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/actuator_test.go]: I see `"machine.openshift.io/cluster-api-machine-role"` used ~4 times - turn it into a const?

Comment (awesomenix) [reply pkg/cloud/azure/actuators/machine/actuator_test.go]: Thanks fixed.
